### DISCUSSION TITLE
Error - Can't Connect to Amazon Athena

### DISCUSSION THREAD

**Discussion by ftironectidata** [dbeaver/dbeaver#21437] [Q&A] [answered]
### Description

[Simba][AthenaJDBC](100071) An error has been thrown from the AWS Athena client. You are not authorized to perform: athena:StartQueryExecution on the resource. After your AWS administrator or you have updated your permissions, please try again. [Execution ID not available]
  You are not authorized to perform: athena:StartQueryExecution on the resource. After your AWS administrator or you have updated your permissions, please try again. (Service: AmazonAthena; Status Code: 400; Error Code: AccessDeniedException; Request ID: acab5009-88d9-499c-b58b-c8159fd892bc; Proxy: null)
  You are not authorized to perform: athena:StartQueryExecution on the resource. After your AWS administrator or you have updated your permissions, please try again. (Service: AmazonAthena; Status Code: 400; Error Code: AccessDeniedException; Request ID: acab5009-88d9-499c-b58b-c8159fd892bc; Proxy: null)

### DBeaver Version

Community Addition 23.2.1.202309260508

### Operating System

Windows 11

### Database and driver

Amazon Athena 

Magnitude Simba Athena JDBC Data
Version 2.0.25

### Steps to reproduce

Try to connect to database

**Comment by LonwoLonwo** [comment]
Hello @ftironectidata 

> You are not authorized to perform: athena:StartQueryExecution on the resource. After your AWS administrator or you have updated your permissions, please try again

Did you try to ask your database admin for special rights?

**Comment by ftironectidata** [comment]
Hi Anastasiya,

Thanks for your help! Please let me know what permissions I need and how to give them.

**Comment by ftironectidata** [comment]
I added the permissions and It’s working now!

Best Regards,

Frank

**Reply by G2RSHappyDog** [reply]
Hello,

We have a user receiving the same error in DBeaver while accessing AWS Athena and I would like to ask what permissions were granted and where in order for the query to successfully run.

Thank you

**Comment by ftironectidata** [comment, marked as answer]
You need permission to perform the `athena:StartQueryExecution` action.

To resolve this issue, follow these steps:

1. **Identify the IAM User or Role**: Determine which IAM user or role you are using to access Athena.

2. **Update IAM Permissions**:
    - Log in to the AWS Management Console.
    - Navigate to the IAM service.
    - Find and select the IAM user or role that needs the updated permissions.
    - Attach or update the policy to include the necessary Athena permissions.

Here's an example of a policy that includes the required permissions:

```json
{
   "Version": "2012-10-17",
   "Statement": [
       {
           "Effect": "Allow",
           "Action": [
               "athena:StartQueryExecution",
               "athena:GetQueryExecution",
               "athena:GetQueryResults"
           ],
           "Resource": "*"
       }
   ]
}
```

3. **Attach the Policy**:
    - If you are using a custom policy, update it to include the above permissions.
    - If you are using managed policies, ensure that the policies attached to your IAM user or role include permissions for Athena.

4. **Verify the Update**:
    - After updating the permissions, attempt to run the query again in Athena to verify that the issue is resolved.

If you do not have the necessary permissions to update IAM roles or policies, contact your AWS administrator for assistance. They will need to grant the appropriate permissions for your user or role.


Best Regards,



Frank



________________________________
